xclim/core/indicator.py

```diff
@@ -853,10 +857,21 @@ def _parse_variables_from_call(self, args, kwds):
                     # If a non-optional variable OR None, store the arg
                     if param.kind == InputKind.VARIABLE or data is not None:
                         das[name] = data
+                        # Retrieve percentile threshold used from percentile_doy result
+                        if "percentile_doy" in data.attrs.get("history", ""):
+                            base_per = data.attrs.get("per_base_thresh")
+                            if len(base_per) == 1:
+                                preformatted_params["per_base_thresh"] = base_per[0]
+                            else:
+                                preformatted_params["per_base_thresh"] = base_per
+                            preformatted_params["per_window"] = data.attrs.get("window")
+                            preformatted_params["per_period"] = data.attrs.get(
+                                "climatology_bounds"
+                            )
```

Comment:
I am not convinced parsing the history is the safest way to check this. I think I'd rather prefer a new `InputKind` entry, that's what they were meant for. The problem here is that a "percentile threshold" is not differentiable from a normal input in the world of `typing`, and thus I think we'll need to use the same workaround as for `DateStr`.

I also believe `per` should be replaced by the input argument's name. Then it would be more intuitive for the user that if the input arg is `t90` then `t90_window` is the extracted window value. This would also allow for doing this on indicators with more than one percentile threshold (like `cold_and_dry_days`). 

Also, I think this code should reside in `format`, in a goal to centralize this parsing and translating.

Reply:
I tend to agree that parsing a string, possibly modified by the user, is particularly unsafe. I went with it because it's already what we do in bootrapping algo.
Another approach could be to inherit DataArray in a PercentileDataArray class which would be outputted by `percentile_doy`. This way it is still a DataArray and can be manipulated as is but, it can as well be recognized as a different `InputKind`.

> Also, I think this code should reside in format, in a goal to centralize this parsing and translating.
I'll try to move these there.

Reply:
This should be resolved by https://github.com/Ouranosinc/xclim/pull/1050/commits/12fab092876292104a9ca9e183c2c8f493fd80b6